templates/tweets/scripts.html

```diff
@@ -0,0 +1,56 @@
+<script type="text/javascript">
+  const getCookie = name => {
+    if (document.cookie && document.cookie !== "") {
+      for (const cookie of document.cookie.split(";")) {
+        const [key, value] = cookie.trim().split("=");
+        if (key === name) {
+          return decodeURIComponent(value);
+        }
+      }
+    }
+  };
+  const csrftoken = getCookie('csrftoken');
+
+
+  const changeStyles = (jsonResponse,func) => {
+    const selector = document.getElementsByName(jsonResponse.tweet_id)[0];
+    if (jsonResponse.liked) {
+      console.log("liked")
+      selector.outerHTML = "<button data-button=\"like\" data-url=" + "\"{% url 'tweets:unlike' 0 %}\"".replace("0", jsonResponse.tweet_id) + "name=\"{{tweet.id}}\"><i class=\"fas fa-lg fa-heart like-red\"></i></button>";
+    } else {
+      console.log("not liked")
+      selector.outerHTML = "<button data-button=\"like\" data-url=" + "\"{% url 'tweets:like' 0 %}\"".replace("0", jsonResponse.tweet_id) + " name=\"{{tweet.id}}\"><i class=\"far fa-lg fa-heart\"></i></button>";
+    }
+    document.getElementsByName(jsonResponse.tweet_id + "-count")[0].textContent = jsonResponse.count;
+    // イベントリスナーの再設定
+    document.getElementsByName(jsonResponse.tweet_id)[0].addEventListener("click", func);
+  }
+
+
+  const likeButtons = document.querySelectorAll('[data-button="like"]');
+  likeButtons.forEach(likeButton => {
+    const tweet_id = likeButton.getAttribute("name");
+    const url = likeButton.dataset.url;
+    const likefunc = function () {
+      fetch(url, {
```

Comment:
まず、urlが根本的に切り替わらない理由は likefunc の外側でurlを宣言している為です。
このため、何度イベントリスナでlikefuncを呼び出しても、すなわち、いいねを押したとしてもurlは切り替わりません。
```suggestion
    const likefunc = function () {
      const tweet_id = likeButton.getAttribute("name");
      const url = likeButton.dataset.url;
      fetch(url, {
```

Reply:
修正しました。